tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -657,17 +657,22 @@ SpirvEvalInfo SPIRVEmitter::loadIfGLValue(const Expr *expr,
       return info.setRValue();
     }
 
-    uint32_t valType = 0;
-    if (valType = info.getValTypeId()) {
+    if (loadIfAliasVarRef(expr, info)) {
       // We are loading an alias variable as a whole here. This is likely for
       // wholesale assignments or function returns. Need to load the pointer.
       //
       // Note: legalization specific code
+      // TODO: It seems we should not set rvalue here since info is still
+      // holding a pointer. But it fails structured buffer assignment because
+      // of double loadIfGLValue() calls if we do not. Fix it.
+      return info.setRValue();
     }
+
+    uint32_t valType = 0;
     // TODO: Ouch. Very hacky. We need special path to get the value type if
     // we are loading a whole ConstantBuffer/TextureBuffer since the normal
     // type translation path won't work.
-    else if (const auto *declContext = isConstantTextureBufferDeclRef(expr)) {
+    if (const auto *declContext = isConstantTextureBufferDeclRef(expr)) {
```

Comment:
else if --> if 
why?

Reply:
The `if` statement will `return` directly anyway.